pkgs/development/beam-modules/elixir-ls/launch.sh.patch

```diff
@@ -0,0 +1,169 @@
+-            . "$ASDF_SH"
+-        else
+-            >&2 echo "Legacy asdf not found at $ASDF_SH"
+-        fi
+-    else
+-        >&2 echo "asdf executable found at $(command -v asdf). Using ASDF_DIR=${ASDF_DIR}, ASDF_DATA_DIR=${ASDF_DATA_DIR}."
+-    fi
+-    export_stdlib_path "asdf which elixir"
+-else
+-    # Fallback to old method for version <= 0.15.x
+-    ASDF_DIR=${ASDF_DIR:-"${HOME}/.asdf"}
+-    ASDF_SH="${ASDF_DIR}/asdf.sh"
+-    if test -f "$ASDF_SH"; then
+-        >&2 echo "Legacy pre v0.16.0 asdf install found at $ASDF_SH, sourcing"
+-        # Source the old asdf.sh script for versions <= 0.15.0
+-        . "$ASDF_SH"
+-        export_stdlib_path "asdf which elixir"
+-    else
+-        >&2 echo "asdf not found"
+-        >&2 echo "Looking for mise executable"
+-
+-        # Look for mise executable
+-        if command -v mise >/dev/null 2>&1; then
+-            >&2 echo "mise executable found at $(command -v mise), activating"
+-            eval "$($(command -v mise) env -s "$preferred_shell")"
+-            export_stdlib_path "mise which elixir"
+-        else
+-            >&2 echo "mise not found"
+-            >&2 echo "Looking for rtx executable"
+-
+-            # Look for rtx executable
+-            if command -v rtx >/dev/null 2>&1; then
+-                >&2 echo "rtx executable found at $(command -v rtx), activating"
+-                eval "$($(command -v rtx) env -s "$preferred_shell")"
+-                export_stdlib_path "rtx which elixir"
+-            else
+-                >&2 echo "rtx not found"
+-                >&2 echo "Looking for vfox executable"
+-
+-                # Look for vfox executable
+-                if command -v vfox >/dev/null 2>&1; then
+-                    >&2 echo "vfox executable found at $(command -v vfox), activating"
+-                    eval "$($(command -v vfox) activate "$preferred_shell")"
+-                else
+-                    >&2 echo "vfox not found"
+-                    export_stdlib_path "which elixir"
+-                fi
+-            fi
+-        fi
+-    fi
+-fi
++#!/usr/bin/env bash
+ 
+ # In case that people want to tweak the path, which Elixir to use, or
+ # whatever prior to launching the language server or the debug adapter, we
+@@ -138,29 +17,18 @@ fi
+ # script so we can correctly configure the Erlang library path to
+ # include the local .ez files, and then do what we were asked to do.
+ 
+-if [ -z "${ELS_INSTALL_PREFIX}" ]; then
+-  SCRIPT=$(readlink_f "$0")
+-  SCRIPTPATH=$(dirname "$SCRIPT")
+-else
+-  SCRIPTPATH=${ELS_INSTALL_PREFIX}
+-fi
++SCRIPT=$(readlink -f "$0")
++SCRIPTPATH=$(dirname "$SCRIPT")/../libexec
+ 
+ export MIX_ENV=prod
+ # Mix.install prints to stdout and reads from stdin
+ # we need to make sure it doesn't interfere with LSP/DAP
+-echo "" | elixir "$SCRIPTPATH/quiet_install.exs" >/dev/null || exit 1
++echo "" | @elixir@/bin/elixir "$SCRIPTPATH/quiet_install.exs" >/dev/null || exit 1
```

Comment:
Ok, this line is weird.

By default, it will use `hex` to fetch the source again, and run with that instead of anything from Nixpkgs' package.

With `ELS_LOCAL=1` it will run the interpreter in the project in `$store/libexec/../`, which will fail.

What exactly are we attempting to do here?

Reply:
This package needs more changes on top of what is in this PR. This was a first pass at cleanup, but this package is indeed weird in that it's not really a package but a set of wrapper scripts. Ideally we properly package it and set ELS_LOCAL so it uses code from the store instead of downloading. Until then I wouldn't expect setting that env var to work.

I've used these changes without issues though. Do you have a reproducer?

Reply:
> I've used these changes without issues though. Do you have a reproducer?

I'm more curious what impurity made it work for you, since there is no codepath that uses the .beam built here, it always uses the one re-downloaded into the mix home path, without the `1216.patch` we need.